interpreter/shared/src/main/scala/sigmastate/types.scala

```diff
@@ -907,13 +908,18 @@ case object SBoolean extends SPrimType with SEmbeddable with SLogical with SProd
   override val reprClass: RClass[_] = RClass(classOf[Boolean])
 
   val ToByte = "toByte"
-  protected override def getMethods() = super.getMethods()
-  /* TODO soft-fork: https://github.com/ScorexFoundation/sigmastate-interpreter/issues/479
-  ++ Seq(
-    SMethod(this, ToByte, SFunc(this, SByte), 1)
-      .withInfo(PropertyCall, "Convert true to 1 and false to 0"),
-  )
-  */
+
+  lazy val ToByteMethod: SMethod = SMethod(
+    this, ToByte, SFunc(this, SByte), 1, FixedCost(JitCost(1)))
+    .withInfo(PropertyCall, "Convert true to 1 and false to 0")
+
+  protected override def getMethods(): Seq[SMethod] = {
+    if (VersionContext.current.isEvolutionActivated) {
```

Comment:
It should not, but I do not see where tree-based version can be injected with the current design, in both serializers and here tree version is not available. So we need to discuss where to inject tree version.

Reply:
It is more or less clear how to make activatedVersion and ergoTreeVersion available in this code place when it is used in the interpreter, I can do this. Because the values are coming from the block being validated. 
But this code is also used in compiler, and there we need to decide, how applications should respect those versions. I.e. if some dApp or tool uses the compiler, how the versions should be specified?

Reply:
I think SigmaByteReader should include tree version. Interpreter is another issue. 

```
 if some dApp or tool uses the compiler, how the versions can be specified
```

this question already exists in the wild, as compiler sets ErgoTree version.